DataProviders/GoogleCredentialProvider.py

```diff
@@ -0,0 +1,21 @@
+import configparser
+import gspread
+from oauth2client.service_account import ServiceAccountCredentials
+
+
+# TODO: Should this a singleton-patterned accessible class to make this more robust?
```

Comment:
If it were me, I'd just do all of the sheets calls through an interface and have the commands call that interface. The stuff that we're doing in `SafeOauth2Command` could as easily just be a wrapper around all of our sheets API calls. This authentication stuff could just live in the same package. You're already doing this, basically, but I'm suggesting a different packaging structure.